.gitignore

```diff
@@ -74,3 +74,7 @@ examples/thermostat/ameba/build
 
 # Downloaded zap without a pigweed root (via zap_download.py)
 .zap
+
+# for idf component manager files
+examples/*/esp32/managed_components
```

Comment:
Why do we expect these to appear? Could we create an out-of-tree build that does not dirty the source tree?

Reply:
There's been a lot of restructuring in esp-idf v5.0, most of non-core component are being moved out to [component registry](https://components.espressif.com/). These components are fetched using [idf-component-manager](https://github.com/espressif/idf-component-manager) and stored at `project/managed_components` making the tree dirty.

I'll check if there's a way to build examples out-of-tree and if I found something I will put up a follow up PR?

Reply:
Could we provide a definitive answer (either find a workaround or exaplain why a workaround is not possible in the comment here)? I am concerned that "if something will be found, a future PR will follow" is likely to result in this not being done.

At least if followup PR seems to be needed (you cannot find something within a reasonable time), please document here some explanation and links about managed_components and why this exists.

Reply:
I have added a comment and a link stating why those files are generated